src/main/java/org/opensearch/knn/index/query/FilterIdsSelector.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.query;
+
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.util.BitSet;
+import org.apache.lucene.util.BitSetIterator;
+import org.apache.lucene.util.FixedBitSet;
+
+import java.io.IOException;
+
+import static org.opensearch.knn.common.KNNConstants.MAX_ID_SELECT_ARRAY;
+
+/**
+ * Util Class for filter ids selector
+ */
+public class FilterIdsSelector {
+
+    /**
+     * When do ann query with filters, there are two types:
+     * BitMap using FixedBitSet, BATCH using a long array stands for filter result docids.
+     */
+    public enum FilterIdsSelectorType {
+
+        BITMAP(0),
+
+        BATCH(1);
+
+        private final int value;
+
+        FilterIdsSelectorType(int value) {
+            this.value = value;
+        }
+
+        public int getValue() {
+            return value;
+        }
+    }
+
+    long[] filterIds;
+    FilterIdsSelectorType filterType;
+
+    public FilterIdsSelector(long[] filterIds, FilterIdsSelectorType type) {
+        this.filterIds = filterIds;
+        this.filterType = type;
+    }
```

Comment:
you can use @AllArgsConstructor annotation here.

Reply:
FIXED at https://github.com/opensearch-project/k-NN/pull/1402/commits/568972fc3dce704bab6dc26b356cb37ef54de1e3